zb-db/src/main/java/io/zeebe/db/impl/rocksdb/transaction/ZeebeTransactionDb.java

```diff
@@ -130,7 +130,11 @@ public void createSnapshot(final File snapshotDir) {
   @Override
   public TransactionContext createContext() {
     final Transaction transaction = optimisticTransactionDB.beginTransaction(defaultWriteOptions);
-    final ZeebeTransaction zeebeTransaction = new ZeebeTransaction(transaction);
+    final ZeebeTransaction zeebeTransaction =
+        new ZeebeTransaction(
+            transaction,
+            oldTransaction ->
```

Comment:
We can create a new lambda every time, or this could be implemented by `ZeebeTransactionDb` since it doesn't really capture any local information (but captures the reference to this instance of `ZeebeTransactionDb`). I have a slight preference for that (e.g. avoiding creating lambdas if possible), but no strong argument so as you want :slightly_smiling_face:

Reply:
Makes sense. Done.